### PR TITLE
9 part 2 extracting config and separating messaging implementations

### PR DESCRIPTION
**Pull request by jbowkett**
Part 2 - extract config & also separate messaging implementations

### CODE REVIEW

File: cassandra-ns/src/main/java/com/excelian/mache/cassandra/builder/CassandraProvisioner.java [hunk 1]
```diff
@@ -1,44 +1,45 @@
...
+    public <K, V> Mache<K, V> getCache(String keySpace, Class<V> valueType, SchemaOptions schemaOption, ClusterDetails clusterDetails, StorageServerDetails... serverDetails) {
+        final Cluster cluster = getCluster(serverDetails[0], clusterDetails);
+        final CassandraCacheLoader<K, V> cacheLoader = getCacheLoader(keySpace, cluster, valueType, schemaOption);
+
+        final MacheFactory macheFactory = new MacheFactory();
+        return macheFactory.create(cacheLoader);
+    }
+
+    private Cluster getCluster(StorageServerDetails server, ClusterDetails clusterDetails) {
+        System.out.println("Connecting to Cassandra cluster...");
```
Comment (tellisnz) [inline cassandra-ns/src/main/java/com/excelian/mache/cassandra/builder/CassandraProvisioner.java]: have you tidied up these sysouts later?

File: cassandra-ns/src/main/java/com/excelian/mache/cassandra/DefaultCassandraConfig.java [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    public String getReplicationClass() {
+        return REPLICATION_CLASS;
+    }
+
+    public ConstantReconnectionPolicy getReconnectionPolicy() {
+        return new ConstantReconnectionPolicy(getReconnectionDelayMs());
+    }
+
+    public static long getReconnectionDelayMs() {
+        return 100L;
```
Comment (tellisnz) [inline cassandra-ns/src/main/java/com/excelian/mache/cassandra/DefaultCassandraConfig.java]: i guess it's not a magic number when it's in a method like this - but then for consistency you've got the others above...

File: cassandra-ns/src/main/java/com/excelian/mache/cassandra/builder/CassandraProvisioner.java (hunk 1 above)
Comment (jbowkett) [inline cassandra-ns/src/main/java/com/excelian/mache/cassandra/builder/CassandraProvisioner.java]: Argh. I did some but have missed some

File: mache-activemq/src/test/java/com/excelian/mache/observable/MessageQueueObservableCacheFactoryIntegrationTest.java [hunk 3]
```diff
@@ -53,14 +57,16 @@ public void beforeTest() throws JMSException {
...
-		mqFactory1 = new ActiveMQFactory(LOCAL_MQ);
+		activeMqConfig = new DefaultActiveMqConfig();
+
+		mqFactory1 = new ActiveMQFactory(LOCAL_MQ, activeMqConfig);
 		observableCacheFactory1 = new MessageQueueObservableCacheFactory(mqFactory1, mqConfiguration, spiedMacheFactory, uuidUtils);
 
 		unspiedCache1 = macheFactory.create(cacheLoader);
-		spiedCache1 = spy(unspiedCache1);
-		when(spiedMacheFactory.create(cacheLoader)).thenReturn(spiedCache1);
+		spiedCache1 = Mockito.spy(unspiedCache1);
```
Comment (tellisnz) [inline mache-activemq/src/test/java/com/excelian/mache/observable/MessageQueueObservableCacheFactoryIntegrationTest.java]: any particular beef against static imports or is this your ide being helpful?

**Comment by tellisnz** [comment]
LGTM after those.

**Comment by alex-smith-excelian** [comment]
Just looking at one of the recent branches. @jbowkett  did you go with my idea to remove the individual config classes for the Messaging services? It looks like the change has been made but can't see the corresponding properties file.

**Comment by jbowkett** [comment]
@alex-smith-excelian not pulled the config out into properties files due to prioritising other bigger refactorings - as this was just the first step in centralising the config into one place.  Extracting to props files probably makes sense, but for a first release, possibly not on the critical path, but we'd need to decide where those files are found on the classpath and how they get overrridden - not hard, just needs a conversation...

**Pull request merged**